lua/care/types/api.lua

```diff
@@ -43,6 +43,13 @@
 --- Allows the index which represents which entry is selected to be directly set.
 --- This allows to jump anywhere in the completion menu.
 ---@field set_index fun(index: integer): nil
+--- @usage ```lua
+--- local has_selection = function()
+---     return require("care").api.get_intex() ~= 0
+--- end
+--- ```
+--- @return The currently selected entry, 0 representing no selection
```

Comment:
@usage and @return isn't something the documentation script will recognize
so you shouldn't use this here

I'd just write at the top what it does (return index of selection), notice that it doesn't return the entry but just an integer
The example is useful so I'd keep that, just write a quick sentence above it what it does

the exact wording is up to you of course
the codeblock you can write there in the annotations just like you have now (without usage thing so `--- ```lua` on the first line)

Reply:
poor guys named return and usage :D
I'm sure with these names it's not the first ping

Reply:
Sure thing, just wanted to squash once thigns are set, so we could trace back if needed on this huge implementation effort :wink: 

I thought its LDoc for docs, brb with the nice fork

Reply:
Well I use a small luascript for docgen
it recognizes just markdown as documentation outside of the `---@field <whatever you want>`